src/backend/distributed/shared_library_init.c

```diff
@@ -689,6 +689,18 @@ RegisterCitusConfigVariables(void)
 		GUC_STANDARD,
 		NULL, NULL, NULL);
 
+	DefineCustomBoolVariable(
+		"citus.disable_drop_and_truncate_on_known_shards",
```

Comment:
Maybe `allow_changes_to_shards`, we may expand the scope to include other DDL commands and perhaps DML in the future.

Reply:
Like disabling deleting or inserting directly to a shard?

Reply:
perhaps, but there are also things like ALTER TABLE which can be even more destructive.

Some other variants:
- allow_changes_to_shards
- enable_changes_to_shards
- allow_manual_changes_to_shards
- allow_changes_to_known_shards

Reply:
I like `allow_manual_changes_to_shards` as it makes it clear that this is about "manual" changes.